src/cli/commands/distinct.rs

```diff
@@ -157,58 +153,35 @@ impl GenericCommand for DistinctCommand {
             _ => unreachable!("Unknown distance metric"),
         };
 
-        let mut colors = match matches.values_of("color") {
+        let fixed_colors = match matches.values_of("color") {
             None => vec![],
             Some(positionals) => {
                 ColorArgIterator::FromPositionalArguments(config, positionals, PrintSpectrum::Yes)
                     .collect::<Result<Vec<_>>>()?
             }
         };
 
-        let fixed_colors = colors.len();
-        if fixed_colors > count {
+        let num_fixed_colors = fixed_colors.len();
+        if num_fixed_colors > count {
             return Err(PastelError::DistinctColorFixedColorsCannotBeMoreThanCount);
         }
 
-        for _ in fixed_colors..count {
-            colors.push(random::strategies::UniformRGB.generate());
-        }
-
-        let mut annealing = SimulatedAnnealing::new(
-            &colors,
-            SimulationParameters {
-                initial_temperature: 3.0,
-                cooling_rate: 0.95,
-                num_iterations: 100_000,
-                opt_target: OptimizationTarget::Mean,
-                opt_mode: OptimizationMode::Global,
-                distance_metric,
-                fixed_colors,
-            },
-        );
-
-        let mut callback: Box<dyn FnMut(&IterationStatistics)> = if verbose_output {
+        let callback: Box<dyn FnMut(&IterationStatistics)> = if verbose_output {
             Box::new(|stats: &IterationStatistics| {
+                let stderr = io::stderr();
+                let brush_stderr = Brush::from_environment(Stream::Stderr);
```

Comment:
why have these lines been moved into the closure? This way we lock `stderr` and create a new brush for every callback iteration.

Reply:
I'm somewhat new to rust and I think that's a good question about what may be an incorrect solution on my part, but where would you put the lock for the output?

I though that here, nearest the output was the most correct. I may be incorrect in my understanding but I imagine handing the lock to the callback from elsewhere would lead to deadlocks.

I'd love to be shown a better, more idiomatic way...

Reply:
We are not running the callback concurrently. If we were, I guess we would actually introduce a potential synchronization overhead if we lock stderr on each iteration. But there is no deadlock scenario here. I modified your PR slightly to lock stderr only once (I didn't really look closely. My original code actually locked stderr multiple times as well, so sorry for that).